src/scancode/cli.py

```diff
@@ -825,11 +819,10 @@ def resource_paths(base_path, pre_scan_plugins=()):
     resources = fileutils.resource_iter(base_path, ignored=ignorer)
 
     for abs_path in resources:
-        posix_path = fileutils.as_posixpath(abs_path)
-        # fix paths: keep the path as relative to the original
-        # base_path. This is always Unicode
-        rel_path = get_relative_path(posix_path, len_base_path, base_is_dir)
-        yield abs_path, rel_path
+        resource = Resource(abs_path, base_is_dir, len_base_path)
+        # always fetch infos and cache.
+        resource.infos.update(scan_infos(abs_path, diag=diag))
```

Comment:
We would need to cache these to avoid recomputing them a second time if info are needed downstream? Or is this taken care of otherwise?